feathr_project/feathr/_feature_registry.py

```diff
@@ -807,3 +817,60 @@ def get_features_from_registry(self, project_name: str, workspace_path: str):
         with open(feature_gen_file, "w") as online_config:
             online_config.write(feature_gen_conf_content)
         logger.info("Writing online configuration from feathr registry to {}", feature_gen_file)
+   
+    def get_feature_by_fqdn_type(self, qualifiedName, typeName):
+        """
+        Get a single feature by it's QualifiedName and Type
+        Returns the feature else throws an AtlasException with 400 error code
+        """
+        response = self.purview_client.get_entity(qualifiedName=qualifiedName, typeName=typeName)
+        entities = response.get('entities')
+        for entity in entities:
+            if entity.get('typeName') == typeName and entity.get('attributes').get('qualifiedName') == qualifiedName: 
+                return entity
+       
+    def get_feature_by_fqdn(self, qualifiedName):
+        """
+        Get feature by qualifiedName
+        Returns the feature else throws an AtlasException with 400 error code
+        """        
+        guid = self.get_feature_guid(qualifiedName)
+        return self.get_feature_by_guid(guid)
+    
+    def get_feature_by_guid(self, guid):
+        """
+        Get a single feature by it's Guid
+        Returns the feature else throws an AtlasException with 400 error code
```

Comment:
why not 404?

Will it ever throw non-400 errors?

Reply:
It will always throw 4xx errors, 404 being one of them. It could be 403 when API code does not match.